.github/workflows/gradle_build.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will compile a Java project with Gradle and run checkstyle + PMD checks
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Compile, Checkstyle, PMD
+
+on:
+  - push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: payitadmin/docker-jenkins-build
+      credentials:
+        username: ${{ secrets.DOCKERHUB_READER_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_READER_API_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Java 13
+        uses: actions/setup-java@v2
+        with:
+          java-version: '13'
+          distribution: adopt
+      - name: Set up environment
+        run: | 
+          echo "ARTIFACTORY_URL=${{ secrets.ARTIFACTORY_URL }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_USERNAME=${{ secrets.ARTIFACTORY_READER_USERNAME }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_PASSWORD=${{ secrets.ARTIFACTORY_READER_PASSWORD }}" >> $GITHUB_ENV
+      - name: Compile
+        run: chmod +x gradlew && ./gradlew clean assemble --refresh-dependencies
```

Comment:
Why clean and refresh-dependencies?

Also, should we compile tests too even though we're not running them?